modepy/matrices.py

```diff
@@ -252,33 +252,68 @@ def __init__(self, face_vertices):
         if npts != vol_dim:
             raise ValueError("face_vertices has wrong shape")
 
-        self.origin = face_vertices[:, 0]
-        self.span = face_vertices[:, 1:] - self.origin[:, np.newaxis]
+        self.origin = face_vertices[:, 0].reshape(-1, 1)
+        self.span = face_vertices[:, 1:] - self.origin
 
         self.face_dim = vol_dim - 1
 
     def __call__(self, points):
-        return (self.origin[:, np.newaxis]
-                + np.einsum("ad,dn->an", self.span, points*0.5 + 0.5))
+        return self.origin + np.einsum("ad,dn->an", self.span, points*0.5 + 0.5)
 
 
-def modal_face_mass_matrix(trial_basis, order, face_vertices, test_basis=None):
+class _HypercubeFaceMap:
+    def __init__(self, face_vertices):
+        """
+        :arg face_vertices: an array of shape ``[dim, npts]``, where *npts*
+            should equal `2**(dim - 1)`.
+        """
+        vol_dim, npts = face_vertices.shape
+        if npts != 2**(vol_dim-1):
+            raise ValueError("face_vertices has wrong shape")
+
+        self.origin = face_vertices[:, 0].reshape(-1, 1)
+        self.span = face_vertices[:, -2:0:-1] - self.origin
```

Comment:
Had to reverse things here too for it to work! (see inducer/meshmode#70)

Reply:
Nevermind, the tests do pass without reversing, but it does flip the vertices in the same way.